src/test/scala/GameServiceTest.scala

```diff
@@ -0,0 +1,96 @@
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funspec.AnyFunSpec
+import org.scalamock.scalatest.MockFactory
+import service.GameServiceImpl
+import service.GameService
+import persistance.GameRepository
+import command.Move
+import command.Move._
+import model.GameResult._
+import persistance.GameStatusEntity
+import service.CpuMoveStrategy
+
+class GameServiceTest extends AnyFunSpec with MockFactory with BeforeAndAfterEach {
+
+  var mockgameRepository: GameRepository = _
+  var mockCpuMoveStrategy: CpuMoveStrategy = _
+  var gameService: GameServiceImpl = _
+
+  override def beforeEach() {
+    mockgameRepository = mock[GameRepository]
+    mockCpuMoveStrategy = mock[CpuMoveStrategy]
+    gameService = new GameServiceImpl(mockgameRepository, mockCpuMoveStrategy)
+  }
+
+  describe("Test play") {
+
+    it("user lose") {
+
+      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Paper)
+      (mockgameRepository.insertGameResult _).expects(Rock, Paper, Lose).once()
+
+      val actualResult = gameService.play(Rock)
+
+      assert(actualResult.gameResult == Lose)
+      assert(actualResult.computerMove == Paper)
+      assert(actualResult.userMove == Rock)
+    }
+
+    it("user win") {
+      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Scissors)
+      (mockgameRepository.insertGameResult _).expects(Rock, Scissors, Win).once()
+
+      val actualResult = gameService.play(Rock)
+
+      assert(actualResult.gameResult == Win)
+      assert(actualResult.computerMove == Scissors)
+      assert(actualResult.userMove == Rock)
+    }
+
+    it("draw case") {
+      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Scissors)
+      (mockgameRepository.insertGameResult _).expects(Scissors, Scissors, Draw).once()
+
+      val actualResult = gameService.play(Scissors)
+
+      assert(actualResult.gameResult == Draw)
+      assert(actualResult.computerMove == Scissors)
+      assert(actualResult.userMove == Scissors)
+    }
+  }
+
+  describe("Test getLastGameResult") {
+    it("should return last insert entry") {
```

Comment:
this is another example. A lot of boilerplate to test that `gameService.getLastGameResult` returns the last item in a list that you are providing via mocking... is this test really helpful? 😕

Reply:
No is not really helpful in that case, I think this like a sort of regression test to prevent that someone change the current behavior.  I'm want to be sure that the returned element is the last element of the ordered list, if the method implementation change and the list fetched is manipulated this test could help to become that changes more safety.